### DISCUSSION TITLE
Unable to flash on stm32f469i_disco

### DISCUSSION THREAD

**Discussion by hafizmqasim** [zephyrproject-rtos/zephyr#67728] [Q&A]
Hi All, 

I am facing an issue to flash on STM32F469I Discovery board. Hello World is build successfully using the command 
west build -p always -b stm32f469i_disco samples\hello_world

but gives following error when **west flash** is executed. 

-- west flash: rebuilding
[1/1] cmd.exe /C "cd /D C:\Users\m3n1iz\zephyrproject\zephyr\bui...le=C:/Users/m3n1iz/zephyrproject/zephyr/build/zephyr/zephyr.elf
-- west flash: using runner openocd
-- runners.openocd: Flashing file: C:/Users/m3n1iz/zephyrproject/zephyr/build/zephyr/zephyr.hex
Open On-Chip Debugger 0.12.0 (2023-10-02) [https://github.com/sysprogs/openocd]
Licensed under GNU GPL v2
libusb1 09e75e98b4d9ea7909e8837b7a3f00dda4589dc3
For bug reports, read
        http://openocd.org/doc/doxygen/bugs.html
Info : The selected transport took over low-level target control. The results might differ compared to plain JTAG/SWD
Info : clock speed 2000 kHz
Error: libusb_open() failed with LIBUSB_ERROR_NOT_SUPPORTED
Info : STLINK V2J35M26 (API v2) VID:PID 0483:374B
Info : Target voltage: 3.259631
Error: SRST error

FATAL ERROR: command exited with status 1: openocd -s 'C:/Users/m3n1iz/zephyrproject/zephyr/boards/arm/stm32f469i_disco\support' -f 'C:/Users/m3n1iz/zephyrproject/zephyr/boards/arm/stm32f469i_disco\support\openocd.cfg' '-c init' '-c targets' -c 'reset init' -c 'flash write_image erase C:/Users/m3n1iz/zephyrproject/zephyr/build/zephyr/zephyr.hex' -c 'reset run' -c shutdown

Can someone guide, what is the issue?

**Comment by FRASTM** [comment]
Hello

2 suggestions:
Did you recently west update  ?
Did you try to upgrade the STLINK  of your target board (to STLINK-V3)

**Comment by hafizmqasim** [comment]
Thanks for your response. 

1. west already updated. 
2. Tried following two utilities but west flash reports same above error message. Unable to connect with board using following two utilities. See the picture of utilities. 
![Screenshot 2024-01-19 124947](https://github.com/zephyrproject-rtos/zephyr/assets/90020502/a698c994-1366-4b78-a579-715512b454d2)

3. Also tried STM32CubeProgrammer but unable to connect with board. 
![image](https://github.com/zephyrproject-rtos/zephyr/assets/90020502/82a2e024-c6ab-4ac8-8e7a-e5a545cf33d5)

4. Finally, Converted the debugger interface from ST-Link to JLINK using "STLinkReflash utility". Used "JFlashLite.exe" to flash HEX file of Zephyr. It connects and flash the image. Hello world works perfectly. However, It is strange 3 ST utilities are not working.
